// vuestic-ui/dist/es/src/composables/useForm/useFormParent-8ac0eddb.mjs
import { provide, computed, ref } from "vue";
import { F as FormServiceKey } from "./consts-bc55152e.mjs";
import { u as useFormChild } from "./useFormChild-28e78c0e.mjs";
const createFormContext = (options) => {
  const fields = ref({});
  return {
    // Vue unwrap ref automatically, but types are not for some reason
    fields: computed(() => Object.values(fields.value)),
    doShowError: computed(() => !options.hideErrors),
    doShowErrorMessages: computed(() => !options.hideErrorMessages),
    doShowLoading: computed(() => !options.hideLoading),
    registerField: (uid, field) => {
      fields.value[uid] = field;
    },
    unregisterField: (uid) => {
      delete fields.value[uid];
    }
  };
};
const useFormParent = (options) => {
  const formContext = createFormContext(options);
  provide(FormServiceKey, formContext);
  const { fields } = formContext;
  const fieldNames = computed(() => fields.value.map((field) => field.name).filter(Boolean));
  const formData = computed(() => fields.value.reduce((acc, field) => {
    if (field.name) {
      acc[field.name] = field.value;
    }
    return acc;
  }, {}));
  const isValid = computed(() => fields.value.every((field) => field.isValid));
  const isLoading = computed(() => fields.value.some((field) => field.isLoading));
  const errorMessages = computed(() => fields.value.map((field) => field.errorMessages).flat());
  const errorMessagesNamed = computed(() => fields.value.reduce((acc, field) => {
    if (field.name) {
      acc[field.name] = field.errorMessages;
    }
    return acc;
  }, {}));
  const validate = () => {
    return fields.value.reduce((acc, field) => {
      return field.validate() && acc;
    }, true);
  };
  const validateAsync = () => {
    return Promise.all(fields.value.map((field) => field.validateAsync())).then((results) => {
      return results.every(Boolean);
    });
  };
  const reset = () => {
    fields.value.forEach((field) => field.reset());
  };
  const resetValidation = () => {
    fields.value.forEach((field) => field.resetValidation());
  };
  const focus = () => {
    var _a;
    (_a = fields.value[0]) == null ? void 0 : _a.focus();
  };
  const focusInvalidField = () => {
    const invalidField = fields.value.find((field) => !field.isValid);
    invalidField == null ? void 0 : invalidField.focus();
  };
  useFormChild(() => ({
    name: "",
    value: void 0,
    isValid: isValid.value,
    isLoading: isLoading.value,
    validate,
    validateAsync,
    reset,
    resetValidation,
    focus,
    errorMessages: errorMessages.value
  }));
  return {
    formData,
    fields,
    fieldNames,
    isValid,
    isLoading,
    errorMessages,
    errorMessagesNamed,
    validate,
    reset,
    resetValidation,
    focus,
    focusInvalidField
  };
};
export {
  useFormParent as u
};
