// vuestic-ui/dist/es/main.js
import { p } from "./src/services/color/presets-dce40c5a.mjs";
import { d } from "./src/services/breakpoint/index-a161fafe.mjs";
import { d as d2 } from "./src/services/global-config/types-95987fc1.mjs";
import { u } from "./src/services/global-config/global-config-d78402d0.mjs";
import { u as u2 } from "./src/composables/useColors-cd2c0b98.mjs";
import { u as u3 } from "./src/composables/useBreakpoint-3fafd10d.mjs";
import { u as u4 } from "./src/composables/useI18n-1b718f5b.mjs";
import { u as u5 } from "./src/composables/useIcon-cccb7a52.mjs";
import { u as u6 } from "./src/composables/useForm/useForm-2d424ff6.mjs";
import { c } from "./src/services/icon/create-icons-config-af07464a.mjs";
import { V } from "./src/services/icon/presets/fonts-35e6ae86.mjs";
import { V as V2 } from "./src/services/icon/presets/vuestic-aliases-f09bb4e3.mjs";
import { G } from "./src/services/global-config/plugin/index-8a8b88ef.mjs";
import { C } from "./src/services/color/plugin/index-322fa67f.mjs";
import { B } from "./src/services/breakpoint/plugin/index-05ae1524.mjs";
import { V as V3 } from "./src/components/va-toast/plugin/index-3bea9d89.mjs";
import { V as V4 } from "./src/components/va-dropdown/plugin/index-89360c2b.mjs";
import { V as V5 } from "./src/components/va-modal/plugin/index-970b6716.mjs";
import { C as C2 } from "./src/services/cache/plugin/index-3da9421a.mjs";
import { C as C3 } from "./src/services/colors-classes/plugin/index-d50dd4f9.mjs";
import { c as c2 } from "./src/services/vue-plugin/create-vuestic/create-vuestic-e21b4a13.mjs";
import { c as c3 } from "./src/services/vue-plugin/create-vuestic/create-vuestic-essential-24f87a9f.mjs";
import { u as u7 } from "./src/composables/useElementTextColor-72885757.mjs";
import { r } from "./src/services/web-components/register-vuestic-web-components-b3e09b05.mjs";
import { r as r2 } from "./src/services/web-components/register-vuestic-web-components-essential-86660074.mjs";
import { V as V6 } from "./src/components/va-fallback/index-c7099275.mjs";
import { V as V7 } from "./src/components/va-accordion/index-c2d30770.mjs";
import { V as V8 } from "./src/components/va-affix/index-c5e5f46b.mjs";
import { V as V9 } from "./src/components/va-alert/index-587e1d48.mjs";
import { V as V10 } from "./src/components/va-app-bar/index-dc176e7c.mjs";
import { V as V11 } from "./src/components/va-aspect-ratio/index-059c82f7.mjs";
import { V as V12 } from "./src/components/va-avatar/index-84aec4b5.mjs";
import { V as V13 } from "./src/components/va-avatar-group/index-57e25f53.mjs";
import { V as V14 } from "./src/components/va-backtop/index-86a47684.mjs";
import { V as V15 } from "./src/components/va-badge/index-e5cc8a8a.mjs";
import { a, V as V16 } from "./src/components/va-breadcrumbs/index-696950bd.mjs";
import { V as V17 } from "./src/components/va-button-dropdown/index-edac19cc.mjs";
import { V as V18 } from "./src/components/va-button-group/index-7562beb4.mjs";
import { V as V19 } from "./src/components/va-button-toggle/index-c71120ea.mjs";
import { V as V20 } from "./src/components/va-button/index-1db585d5.mjs";
import { d as d3, b, c as c4, V as V21, a as a2 } from "./src/components/va-card/index-8f0cf1cc.mjs";
import { V as V22 } from "./src/components/va-carousel/index-0815012c.mjs";
import { V as V23 } from "./src/components/va-checkbox/index-9272786f.mjs";
import { V as V24 } from "./src/components/va-chip/index-1fea3cc5.mjs";
import { V as V25 } from "./src/components/va-collapse/index-1287247d.mjs";
import { V as V26 } from "./src/components/va-color-indicator/index-38941117.mjs";
import { V as V27 } from "./src/components/va-color-input/index-e2fe9630.mjs";
import { V as V28 } from "./src/components/va-color-palette/index-566dfe86.mjs";
import { V as V29 } from "./src/components/va-config/VaConfig-785c45a1.mjs";
import { V as V30 } from "./src/components/va-content/index-965a86f2.mjs";
import { V as V31 } from "./src/components/va-counter/index-bd17b838.mjs";
import { V as V32 } from "./src/components/va-data-table/index-f955ff1f.mjs";
import { V as V33 } from "./src/components/va-date-input/index-7fd0d868.mjs";
import { V as V34 } from "./src/components/va-date-picker/index-81b2c444.mjs";
import { V as V35 } from "./src/components/va-divider/index-ff85d97f.mjs";
import { V as V36 } from "./src/components/va-dropdown/components/VaDropdownContent/index-0209c7a0.mjs";
import { V as V37 } from "./src/components/va-dropdown/index-9e0e771a.mjs";
import { V as V38 } from "./src/components/va-form/index-66ecb023.mjs";
import { V as V39 } from "./src/components/va-hover/index-75c7258c.mjs";
import { V as V40 } from "./src/components/va-icon/index-7fefcddb.mjs";
import { V as V41 } from "./src/components/va-image/index-c84f2dd1.mjs";
import { V as V42 } from "./src/components/va-infinite-scroll/index-9cdc3ef5.mjs";
import { V as V43 } from "./src/components/va-inner-loading/index-5ae3376b.mjs";
import { V as V44 } from "./src/components/va-input/components/VaInputWrapper/VaInputWrapper-911e995d.mjs";
import { V as V45 } from "./src/components/va-input/components/VaMessageList/index-3de9699a.mjs";
import { V as V46 } from "./src/components/va-input/components/VaMessageListWrapper-789238cc.mjs";
import { V as V47 } from "./src/components/va-input/index-a46ef122.mjs";
import { b as b2, V as V48, d as d4, a as a3, c as c5, e } from "./src/components/va-list/index-d370814a.mjs";
import { u as u8 } from "./src/components/va-modal/hooks/useModal-51c3010b.mjs";
import { V as V49 } from "./src/components/va-modal/VaModal-b1dd8919.mjs";
import { V as V50, a as a4 } from "./src/components/va-navbar/index-15641e4a.mjs";
import { V as V51 } from "./src/components/va-option-list/index-0392c22b.mjs";
import { V as V52 } from "./src/components/va-pagination/index-dbd9e257.mjs";
import { V as V53 } from "./src/components/va-parallax/index-f363fb72.mjs";
import { V as V54 } from "./src/components/va-popover/index-862e71a5.mjs";
import { V as V55 } from "./src/components/va-progress-bar/index-ef09d849.mjs";
import { V as V56 } from "./src/components/va-progress-circle/index-ad19302c.mjs";
import { V as V57 } from "./src/components/va-radio/index-b41b3ca0.mjs";
import { V as V58 } from "./src/components/va-rating/components/VaRatingItem/index-7ef5d3af.mjs";
import { V as V59 } from "./src/components/va-rating/index-c19356b5.mjs";
import { V as V60 } from "./src/components/va-select/components/VaSelectOptionList/index-8fe5b5c4.mjs";
import { V as V61 } from "./src/components/va-select/components/VaSelectOption/index-34a50772.mjs";
import { V as V62 } from "./src/components/va-select/index-ac0feb4c.mjs";
import { V as V63 } from "./src/components/va-separator/VaSeparator-441efce1.mjs";
import { V as V64, a as a5 } from "./src/components/va-skeleton/index-4e0faf13.mjs";
import { V as V65, a as a6, b as b3 } from "./src/components/va-sidebar/VaSidebarItem/index-53fbc5ec.mjs";
import { V as V66 } from "./src/components/va-sidebar/index-4f84abd2.mjs";
import { V as V67 } from "./src/components/va-slider/index-ec4337fa.mjs";
import { V as V68 } from "./src/components/va-spacer/VaSpacer-c6cc3351.mjs";
import { V as V69 } from "./src/components/va-split/index-9ef6d1e5.mjs";
import { V as V70 } from "./src/components/va-stepper/index-a46dc64e.mjs";
import { V as V71 } from "./src/components/va-switch/index-0aad792f.mjs";
import { u as u9 } from "./src/components/va-toast/hooks/useToast-090a10c9.mjs";
import { V as V72 } from "./src/components/va-toast/index-5990aca3.mjs";
import { V as V73 } from "./src/components/va-tabs/components/VaTab/index-2fd17abd.mjs";
import { T } from "./src/components/va-tabs/types-443f9ff0.mjs";
import { V as V74 } from "./src/components/va-tabs/index-33fdc16a.mjs";
import { V as V75 } from "./src/components/va-time-input/index-e6dd8bf1.mjs";
import { V as V76 } from "./src/components/va-time-picker/index-5b3d453f.mjs";
import { V as V77 } from "./src/components/va-timeline/VaTimelineItem/index-a97b2090.mjs";
import { V as V78 } from "./src/components/va-timeline/VaTimelineSeparator/index-b6d817fe.mjs";
import { V as V79 } from "./src/components/va-timeline/index-7194db84.mjs";
import { V as V80 } from "./src/components/va-tree-view/index-240447e0.mjs";
import { V as V81 } from "./src/components/va-scroll-container/index-3beee7bb.mjs";
import { V as V82 } from "./src/components/va-viewer/index-c76589e8.mjs";
import { V as V83 } from "./src/components/va-virtual-scroller/index-e610df5f.mjs";
import { V as V84 } from "./src/components/va-value/index-f4d83597.mjs";
import { V as V85 } from "./src/components/va-file-upload/VaFileUploadGalleryItem/index-79148517.mjs";
import { V as V86 } from "./src/components/va-file-upload/VaFileUploadList/index-6105ec0a.mjs";
import { V as V87 } from "./src/components/va-file-upload/VaFileUploadListItem/index-dbcb2d22.mjs";
import { V as V88 } from "./src/components/va-file-upload/VaFileUploadSingleItem/index-862d7c2d.mjs";
import { V as V89 } from "./src/components/va-file-upload/VaFileUploadUndo/index-718ae37f.mjs";
import { V as V90 } from "./src/components/va-file-upload/types-8c5c5611.mjs";
import { V as V91 } from "./src/components/va-file-upload/index-18324a82.mjs";
export {
  B as BreakpointConfigPlugin,
  C2 as CachePlugin,
  C as ColorConfigPlugin,
  C3 as ColorsClassesPlugin,
  G as GlobalConfigPlugin,
  T as TabsViewKey,
  V7 as VaAccordion,
  V8 as VaAffix,
  V9 as VaAlert,
  V10 as VaAppBar,
  V11 as VaAspectRatio,
  V12 as VaAvatar,
  V13 as VaAvatarGroup,
  V14 as VaBacktop,
  V15 as VaBadge,
  a as VaBreadcrumbs,
  V16 as VaBreadcrumbsItem,
  V20 as VaButton,
  V17 as VaButtonDropdown,
  V18 as VaButtonGroup,
  V19 as VaButtonToggle,
  d3 as VaCard,
  b as VaCardActions,
  c4 as VaCardBlock,
  V21 as VaCardContent,
  a2 as VaCardTitle,
  V22 as VaCarousel,
  V23 as VaCheckbox,
  V24 as VaChip,
  V25 as VaCollapse,
  V26 as VaColorIndicator,
  V27 as VaColorInput,
  V28 as VaColorPalette,
  V29 as VaConfig,
  V30 as VaContent,
  V31 as VaCounter,
  V32 as VaDataTable,
  V33 as VaDateInput,
  V34 as VaDatePicker,
  V35 as VaDivider,
  V37 as VaDropdown,
  V36 as VaDropdownContent,
  V4 as VaDropdownPlugin,
  V6 as VaFallback,
  V91 as VaFileUpload,
  V85 as VaFileUploadGalleryItem,
  V90 as VaFileUploadKey,
  V86 as VaFileUploadList,
  V87 as VaFileUploadListItem,
  V88 as VaFileUploadSingleItem,
  V89 as VaFileUploadUndo,
  V38 as VaForm,
  V39 as VaHover,
  V40 as VaIcon,
  V41 as VaImage,
  V42 as VaInfiniteScroll,
  V43 as VaInnerLoading,
  V47 as VaInput,
  V44 as VaInputWrapper,
  b2 as VaList,
  V48 as VaListItem,
  d4 as VaListItemLabel,
  a3 as VaListItemSection,
  c5 as VaListLabel,
  e as VaListSeparator,
  V45 as VaMessageList,
  V46 as VaMessageListWrapper,
  V49 as VaModal,
  V5 as VaModalPlugin,
  V50 as VaNavbar,
  a4 as VaNavbarItem,
  V51 as VaOptionList,
  V52 as VaPagination,
  V53 as VaParallax,
  V54 as VaPopover,
  V55 as VaProgressBar,
  V56 as VaProgressCircle,
  V57 as VaRadio,
  V59 as VaRating,
  V58 as VaRatingItem,
  V81 as VaScrollContainer,
  V62 as VaSelect,
  V61 as VaSelectOption,
  V60 as VaSelectOptionList,
  V63 as VaSeparator,
  V66 as VaSidebar,
  V65 as VaSidebarItem,
  a6 as VaSidebarItemContent,
  b3 as VaSidebarItemTitle,
  V64 as VaSkeleton,
  a5 as VaSkeletonGroup,
  V67 as VaSlider,
  V68 as VaSpacer,
  V69 as VaSplit,
  V70 as VaStepper,
  V71 as VaSwitch,
  V73 as VaTab,
  V74 as VaTabs,
  V75 as VaTimeInput,
  V76 as VaTimePicker,
  V79 as VaTimeline,
  V77 as VaTimelineItem,
  V78 as VaTimelineSeparator,
  V72 as VaToast,
  V3 as VaToastPlugin,
  V80 as VaTreeView,
  V84 as VaValue,
  V82 as VaViewer,
  V83 as VaVirtualScroller,
  V2 as VuesticIconAliases,
  V as VuesticIconFonts,
  p as colorsPreset,
  c as createIconsConfig,
  c2 as createVuestic,
  c3 as createVuesticEssential,
  d2 as defineVuesticConfig,
  r as registerVuesticWebComponents,
  r2 as registerVuesticWebComponentsEssential,
  d as thresholdsPreset,
  u3 as useBreakpoint,
  u2 as useColors,
  u7 as useElementTextColor,
  u6 as useForm,
  u as useGlobalConfig,
  u4 as useI18nConfig,
  u5 as useIcons,
  u8 as useModal,
  u9 as useToast
};
